roles/openshift_storage_glusterfs/tasks/main.yml

```diff
@@ -135,48 +113,60 @@
 
 - include: heketi_deploy_part1.yml
-  when: openshift_storage_glusterfs_heketi_is_native and openshift_storage_glusterfs_heketi_deploy_is_missing and openshift_storage_glusterfs_heketi_is_missing
+  when:
+  - openshift_storage_glusterfs_heketi_is_native
+  - openshift_storage_glusterfs_heketi_deploy_is_missing
+  - openshift_storage_glusterfs_heketi_is_missing
 
 - name: Determine heketi URL
   oc_obj:
-    kubeconfig: "{{ mktemp.stdout }}/admin.kubeconfig"
     namespace: "{{ openshift_storage_glusterfs_namespace }}"
     state: list
     kind: ep
     selector: "glusterfs in (deploy-heketi-service, heketi-service)"
   register: heketi_url
-  until: "heketi_url.results.results[0]['items'][0].subsets[0].addresses[0].ip != '' and heketi_url.results.results[0]['items'][0].subsets[0].ports[0].port != ''"
+  until:
+  - "heketi_url.results.results[0]['items'][0].subsets[0].addresses[0].ip != ''"
+  - "heketi_url.results.results[0]['items'][0].subsets[0].ports[0].port != ''"
   delay: 10
   retries: "{{ (openshift_storage_glusterfs_timeout / 10) | int }}"
-  when: openshift_storage_glusterfs_heketi_is_native and openshift_storage_glusterfs_heketi_url is undefined
+  when:
+  - openshift_storage_glusterfs_heketi_is_native
+  - openshift_storage_glusterfs_heketi_url is undefined
 
 - name: Set heketi URL
   set_fact:
     openshift_storage_glusterfs_heketi_url: "{{ heketi_url.results.results[0]['items'][0].subsets[0].addresses[0].ip }}:{{ heketi_url.results.results[0]['items'][0].subsets[0].ports[0].port }}"
-  when: openshift_storage_glusterfs_heketi_is_native and openshift_storage_glusterfs_heketi_url is undefined
+  when:
+  - openshift_storage_glusterfs_heketi_is_native
+  - openshift_storage_glusterfs_heketi_url is undefined
 
 - name: Verify heketi service
-  uri:
-    url: "http://{{ openshift_storage_glusterfs_heketi_url }}/hello"
-    return_content: True
-  register: heketi_hello
-  failed_when: "'Hello from Heketi' not in heketi_hello.content"
+  command: "heketi-cli -s http://{{ openshift_storage_glusterfs_heketi_url }} --user admin --secret '{{ openshift_storage_glusterfs_heketi_admin_key }}' cluster list"
   changed_when: False
 
 - name: Generate topology file
   template:
     src: "{{ openshift.common.examples_content_version }}/topology.json.j2"
     dest: "{{ mktemp.stdout }}/topology.json"
-  when: openshift_storage_glusterfs_is_native and openshift_storage_glusterfs_heketi_topology_load
+  when:
+  - openshift_storage_glusterfs_is_native
+  - openshift_storage_glusterfs_heketi_topology_load
 
 - name: Load heketi topology
-  command: "heketi-cli -s http://{{ openshift_storage_glusterfs_heketi_url }} topology load --json={{ mktemp.stdout }}/topology.json 2>&1"
+  command: "heketi-cli -s http://{{ openshift_storage_glusterfs_heketi_url }} --user admin --secret '{{ openshift_storage_glusterfs_heketi_admin_key }}' topology load --json={{ mktemp.stdout }}/topology.json 2>&1"
```

Comment:
You may have noted earlier, but if the endpoint could be over TLS it may be nice to store the variable as the full URL.